BLSQ/iaso
# POLIO-1353 Fixes on Supply Chain and Stock Management

This is the "big" branch with all the latest changes on Polio Vaccine Supply Chain and Stock Management.

Related JIRA tickets : POLIO-1353 , POLIO-1147 , POLIO-1332, POLIO-1334

This also includes work from Son on tickets : POLIO-1328 ,  POLIO-1329 , POLIO-1330 , POLIO-1395 , POLIO-1338

Some of the changes :

### Supply Chain

- [x] In supply chain, the filter on PO doesn't work.
- [x] Doses per vials is not editable anymore
- [x] Add fields for vials_shipped, vials_received on VaccineArrivalReport and the model will calculate it by himself on save.

### Stock Management

##### Page List
- [x] Remove % leftover on quantities ordered.
- [x] Missing filters et l'ordering

Many other small improvements here and there.
Catching up with main.
Also fixed the current (not many) tests for stock management and supply chain

## Self proofreading checklist


## Changes

A lot of changes on the backend and frontend of Supply Chain and Stock management.
See above.


## How to test

You can use the frontend to see Vaccine Module then Supply Chain et Stock Management.
You will need a lot of data for it all to work.
Check `setUpTestData` in ` plugins.polio.tests.test_vaccine_stock_management.VaccineStockManagementAPITestCase` for an overview of all the required objects.
But Basically you will Need :
A `Country`, A `Campaign`, A `Campaign Round`.
A `VaccineRequestForm` for this campaign and vaccine type. 
Then a `VaccineArrivalReport` for this request form.
Then create a `VaccineStock` for the Country/Account/Campaign triplet.
Then create some `OutgoingStockMovement`, `DestructionReport` and `IncidentReport`.

(Some of which you can create from the frontend, some you need to go through django admin still)
